### PR TITLE
Scroll year chart in toolbar

### LINKED ISSUE #340: Scroll toolbar's year chart
In continuation to the work done in PR #332, I suggest that scrolling year chart which is in toolbar appearing in every archived webpage will complete the work which has begun in issue #214 and issue #328.

### PR DESCRIPTION
**Pull request by MohammedElsayyed**
This fixes #340. It is now open for code review.

### CODE REVIEW

**Comment by MohammedElsayyed** [comment]
It has been deployed on BA cluster for testing.
http://web.archive.bibalex.org/web/20020213015417/http://www.cnn.com/

**Comment by ldko** [comment]
With this change, when using the next/previous navigation in the toolbar to move between captures, after clicking an arrow to another capture, the next/prev capture navigation and day of month indicator as well as the new dotted scrollbar do not appear and the full length of the capture chart is visible until the content of the archived page finishes loading. This causes a somewhat awkward looking transition. Is it possible to make a smoother transition?

Before clicking blue navigation arrow:
nav

While page is loading:
chart_over_nav

**Comment by MohammedElsayyed** [comment]
Thanks @ldko! It has been fixed. Can you check it now and let me know your findings? 
If everything is going well, then I will update PR.

**Comment by ldko** [comment]
It looks smoother. Please update the PR and I will take a look and deploy locally. Thank you!

**Comment by MohammedElsayyed** [comment]
While I am testing it one more time after committing such change, it sounds like scroll-dragger is resetting after refreshing an archived webpage or going from one page to another.

To try this test case, just go to any webpage, scroll the scroll-dragger e.g. to be in the middle of year chart, and refresh the page.
Expected result is not changing its location, it should keep track of its last location before refreshing the page and update it every time user reloading it. But it resets!

I already fixed this issue, but it seems the last commit changed the normal behavior. Will fix it and update the PR.

**Comment by MohammedElsayyed** [comment]
It has been fixed now! Feel free to give it a try at BA's OWB or locally and let me know your findings.
I am sorry for any difficulty in accessing our resource store, we are losing many disks in last few months. We are recovering them as soon as we can.

File: wayback-webapp/src/main/webapp/WEB-INF/replay/Toolbar.jsp [hunk 1]
```diff
@@ -284,7 +287,8 @@ function trackMouseMove(event,element) {
        </td>
        <td style="padding:0!important;">
        <a style="position:relative; white-space:nowrap; width:<%= imgWidth %>px;height:<%= imgHeight %>px;" href="" id="wm-graph-anchor">
```
Comment (MohammedElsayyed) [inline wayback-webapp/src/main/webapp/WEB-INF/replay/Toolbar.jsp]: I think there will be no difference whether it has been added inside the yearChart or outside it. The behavior is the same. The functionality of the element with id="wm-graph-anchor" is just highlighting the year when mouse pointer is hovering over it.

I am afraid of not fully understand the strange behavior.

File: wayback-webapp/src/main/webapp/WEB-INF/replay/Toolbar.jsp [hunk 2]
```diff
@@ -320,6 +325,30 @@ function trackMouseMove(event,element) {
...
 </div>
+   
+<script>
+    var x = sessionStorage.getItem("scrollX");
+    
+    (function($){
+        $("#yearChart").mCustomScrollbar({
+            axis: "x",
+            theme: "rounded-dots-dark",
+            setLeft: x,
```
Comment (MohammedElsayyed) [inline wayback-webapp/src/main/webapp/WEB-INF/replay/Toolbar.jsp]: Yes, it should be removed. There is no need to set scroll-dragger here. It has been set inside callback function instead. I will update PR here and will fix it in BubbleCalendar too if needed.

File: wayback-webapp/src/main/webapp/WEB-INF/replay/Toolbar.jsp (hunk 1 above)
Comment (ldko) [inline wayback-webapp/src/main/webapp/WEB-INF/replay/Toolbar.jsp]: When the `<a>` is outside of the `yearChart` (as it is in this PR) the `wm-graph-anchor` is being applied to `mCSB_1_scrollbar_horizontal` as well as `wm-ipp-sparkline`. 

broader_anchor_markup

When the `<a>` is inside of the `yearChart` `<div>` it is only applied to `wm-ipp-sparkline`.

narrower_anchor_markup

The dotted scrollbar ends up linked too--which I don't think is the desired behavior. The problem this is causing for me is that then when I scroll or click on the dotted scrollbar, it is linked and can take me to a different capture. If I have previously had the mouse pointer hovering over the year chart and the <a> href is now for a different date, I go to that date when I move the scrollbar any direction and then release.

Comment (MohammedElsayyed) [inline wayback-webapp/src/main/webapp/WEB-INF/replay/Toolbar.jsp]: Yes, you are absolutely right, @ldko. 
Thanks for your superb review!

**Comment by ldko** [comment]
Thanks for updating this, @MohammedElsayyed. It is working well for me now. Since this is a new feature, I would like to do a 2.3.2 tag/release before merging this and then put this on 2.4.0.

**Pull request merged**